dpctl/_sycl_device.pyx

```diff
@@ -639,6 +643,22 @@ cdef class SyclDevice(_SyclDevice):
             self._device_ref
         )
 
+    @property
+    def platform(self):
+        """ Returns the platform associated with this device.
+
+        Returns:
+            :class:`dpctl.SyclPlatform`: The platform associated with this
+                device.
+        """
+        cdef DPCTLSyclPlatformRef PRef = (
+            DPCTLDevice_GetPlatform(self._device_ref)
+            )
+        if (PRef == NULL):
+            raise RuntimeError("Could not get platform for device.")
+        else:
+            return SyclPlatform._create(PRef)
```

Comment:
I am assuming `_create` will consume the `PRef`?

Reply:
Yes, it does.